backfill_corrections/Makefile

```diff
@@ -33,19 +54,41 @@ run-local:
 	[ "$$?" -eq 1 ]
 
 run:
-	# Assumes receiving is output dir
 	docker run --rm --pull=always \
-	    -v "${PWD}"/logs:/backfill_corrections/logs \
-	    -v "${PWD}"/receiving:/backfill_corrections/receiving \
-	    -v "${PWD}"/input:/backfill_corrections/input \
-	    -v "${PWD}"/cache:/backfill_corrections/cache \
+	    -v "${PWD}/${LOG_DIR}:/backfill_corrections/${LOG_DIR}" \
+	    -v "`realpath $(USR_EXPORT_DIR)`:/backfill_corrections/${EXPORT_DIR}" \
+	    -v "`realpath $(USR_INPUT_DIR)`:/backfill_corrections/${INPUT_DIR}" \
+	    -v "`realpath $(USR_CACHE_DIR)`:/backfill_corrections/${CACHE_DIR}" \
 	    -v "${PWD}"/params.json:/backfill_corrections/params.json \
 		-i "${DOCKER_IMAGE}:${DOCKER_TAG}" \
 		/bin/bash -c "make run-local OPTIONS=\"${OPTIONS}\""
 
-deploy: run
-	# Assumes receiving is output dir
-	aws s3 cp receiving/ $(S3_BUCKET)/ --recursive --exclude "*" --include "*csv.gz" --acl public-read
+publish:
+ 	aws configure set aws_access_key_id AWS_KEY_ID
+ 	aws configure set aws_secret_access_key AWS_SECRET_KEY
+ 	aws s3 cp $(USR_INPUT_DIR) $(S3_BUCKET)/ --recursive --exclude "*" --include "*.csv.gz" --acl public-read
+ 	echo "SUCCESS: published `ls -1 $(USR_EXPORT_DIR)/*.csv.gz | wc -l` files to the S3 bucket" >> $(LOG_FILE)
+
+pipeline: setup run publish teardown
+
+# Make sure all required dirs exist. Set dir names to standard values for use in Docker.
+setup:
+	[ -f $(USR_INPUT_DIR) ] || mkdir -p $(USR_INPUT_DIR)
+	[ -f $(USR_CACHE_DIR) ] || mkdir -p $(USR_CACHE_DIR)
+	[ -f $(USR_EXPORT_DIR) ] || mkdir -p $(USR_EXPORT_DIR)
```

Comment:
...are we ensuring these directories exist, then immediately configuring params.json to use other directories?

Reply:
Yep. Definitely open to suggestions here, but let me explain my thinking.

* User provides path to dirs in `params`. Sometimes dirs can be outside of the working dir. This is especially true of the input dir, which at the moment we've fixed as `/common/backfill/<indicator name>`.
* Check that dirs exist locally and create them if not. This is because we need to mount them in the docker container. If we attempt to mount a dir that doesn't actually exist locally, docker will happily use it but fail to copy the contents (output, in this case) back to the host.
* Since the dirs inside docker are obscured, I figured it would be easier if they just had fixed names/locations hence reconfiguring `params`

Reply:
But... I guess I can just use the user-provided names inside the container directly? They won't necessarily end up in the working directory, but that should be fine.

Reply:
An additional complication with using user-provided dir names inside the container is that we need different processing if the dir name in `params` is an absolute or a relative path. This is because when [mounting the dirs in `docker run`](https://github.com/cmu-delphi/covidcast-indicators/blob/6c140016f48045599a137c240daecd976997a1d4/backfill_corrections/Makefile#L59) we would need to specify the parent dir for relative paths.

Instead of checking if dir paths from `params` are absolute or not, we will keep the current approach and document the choice. The two approaches are equally finicky, but the current approach has been tested already.

Reply:
Addressed in https://github.com/cmu-delphi/covidcast-indicators/pull/1707/commits/bbbb59f370dec0bdeb0c7ce836813b620544fa22